js/l1_player.js

```diff
@@ -0,0 +1,274 @@
+        w.player.unmute();
+      });
+    },
+    toggleMute() {
+      backgroundCall((w) => {
+        if (w.player.muted) w.player.unmute();
+        else w.player.mute();
+      });
+    },
+    volume(per) {
+      backgroundCall((w) => {
+        const { player } = w;
+        player.volume = per / 100;
+      });
+    },
+    adjustVolume(increase) {
+      backgroundCall((w) => {
+        w.player.adjustVolume(increase);
+      });
+    },
+    addTrack(track) {
+      backgroundCall((w) => {
+        w.player.insertAudio(track);
+      });
+    },
+    removeTrack(index) {
+      backgroundCall((w) => {
+        w.player.removeAudio(index);
+      });
+    },
+    addTracks(list) {
+      backgroundCall((w) => {
+        w.player.appendAudioList(list);
+      });
+    },
+    clearPlaylist() {
+      backgroundCall((w) => {
+        w.player.clearPlaylist();
+      });
+    },
+    setNewPlaylist(list) {
+      backgroundCall((w) => {
+        w.player.setNewPlaylist(list);
+      });
+    },
+    getTrackById(id) {
+      if (!l1Player.status.playlist) return null;
+      return l1Player.status.playlist.find(track => track.id === id);
+    },
+    setBootstrapTrack(fn) {
+      l1Player.bootstrapTrack = fn;
+    },
+    connectPlayer() {
+      const player = this;
+      backgroundCall((w) => {
+        w.player.sendFullUpdate();
+        w.player.sendPlaylistEvent();
+        w.player.sendPlayingEvent();
+        w.player.sendLoadEvent();
+        if (!w.player.playing) {
+          initPlayer(player);
+        }
+      });
+    },
+  };
+
+  l1Player.injectDirectives = (ngApp) => {
+    ngApp.directive('playFromPlaylist', () => ({
+      restrict: 'EA',
+      scope: {
+        song: '=playFromPlaylist',
+      },
+      link(scope, element) {
+        element.bind('click', () => {
+          l1Player.playById(scope.song.id);
+        });
+      },
+    }));
+
+    ngApp.directive('nextTrack', () => ({
+      restrict: 'EA',
+      link(scope, element) {
+        element.bind('click', () => {
+          l1Player.next();
+        });
+      },
+    }));
+
+    ngApp.directive('prevTrack', () => ({
+      restrict: 'EA',
+      link(scope, element) {
+        element.bind('click', () => {
+          l1Player.prev();
+        });
+      },
+    }));
+
+    ngApp.directive('clearPlaylist', () => ({
+      restrict: 'EA',
+      link(scope, element) {
+        element.bind('click', () => {
+          l1Player.clearPlaylist();
+        });
+      },
+    }));
+
+    ngApp.directive('removeFromPlaylist', () => ({
+      restrict: 'EA',
+      scope: {
+        song: '=removeFromPlaylist',
+      },
+      link(scope, element, attrs) {
+        element.bind('click', () => {
+          l1Player.removeTrack(attrs.index);
+        });
+      },
+    }));
+
+    ngApp.directive('playPauseToggle', () => ({
+      restrict: 'EA',
+      link(scope, element) {
+        element.bind('click', () => {
+          l1Player.toggleplayPause();
```

Comment:
```suggestion
          l1Player.togglePlayPause();
```
it seems that togglePlayPause is still underscore. Causing the play button not working.